include/swift/SIL/SILDebugScope.h

```diff
@@ -36,18 +37,20 @@ class SILInstruction;
 /// debug info. In contrast to LLVM IR, SILDebugScope also holds all
 /// the inlining information. In LLVM IR the inline info is part of
 /// DILocation.
-class SILDebugScope : public SILAllocated<SILDebugScope> {
+class SILDebugScope final
+    : public SILAllocated<SILDebugScope>,
+      private llvm::TrailingObjects<SILDebugScope, const SILDebugScope *> {
```

Comment:
I'm afraid that the majority of all SILDebugScopes in an optimized stdlib will be of the inlined kind, but this is probably still worth it :-)

Reply:
I agree, but still a nice improvement